File: examples/java-gradle/src/main/java/com/example/interfaces/JsonSerializable.java
package com.example.interfaces;

/**
 * Interface for JSON serialization support.
 */
public interface JsonSerializable {

    /**
     * Convert object to JSON string.
     *
     * @return JSON representation
     */
    String toJson();

    /**
     * Pretty print JSON.
     *
     * @return formatted JSON string
     */
    default String toPrettyJson() {
        return toJson();
    }
}

File: examples/kotlin-gradle/src/main/kotlin/com/example/interfaces/JsonSerializable.kt
package com.example.interfaces

/**
 * Interface for JSON serialization support.
 */
interface JsonSerializable {

    /**
     * Convert object to JSON string.
     */
    fun toJson(): String

    /**
     * Pretty print JSON.
     */
    fun toPrettyJson(): String = toJson()
}
